### ISSUE TITLE
`img` tag incorrectly subject to URL replacement

### ISSUE THREAD

**Issue by ilikepi** [hipchat/hipchat-cli#20]
According to the Hipchat API docs, the `img` tag is supported in html-formatted notification messages.  Unfortunately, if the message input includes an `img` tag, this script will attempt to convert the value of the `src` attribute to an HTML anchor element.  This is obviously undesirable, as it results in the posting of a malformed message.

The simplest fix would involve a tweak to the URL replacement regex to prevent it from acting on `img` tags.  The following should suffice (though it could certainly be made more terse):

``` bash
# replace bare URLs with real hyperlinks
INPUT=$(echo -n "${INPUT}" | perl -p -e "s/(?<!href=\"|href='|src=\"|src=')((?:https?|ftp|mailto)\:\/\/[^ \n]*)/\<a href=\"\1\"\>\1\<\/a>/g")
```

I'm happy to submit a patch to accomplish this, but I wanted to suggest a possible alternative first.  I'm wondering if perhaps the URL replacement should be disabled for html-formatted messages.  It's definitely helpful for text-formatted messages, but it seems more intuitive that an html-formatted message should be required to build links explicitly.

This would be a breaking change of course, but I'm not sure whether that is a concern for the use case of this script.

**Comment by jimsmith** [comment]
more than 2 years old, if not getting progressed then close ?

**Comment by ilikepi** [comment]
@jimsmith This repo hasn't been touched in over 2 years.  If Atlassian isn't going to make further contributions to it, that's their decision.  There is still value in having the repo exist as a starting point to others who need to use the functionality provided by the script.  For the sake of those people, it makes sense to me to keep the existing issues open in order to highlight them.

**Comment by dkolbly** [comment]
@ilikepi As much as I like the cleaner alternative of just stating that html-formatted messages are, well, HTML-formatted (i.e., regarding it as a bug that we mark up URLs in supposedly already HTML messages), I'm worried about the breakage that might cause.  I think some people use the HTML mode to do inline markup like `<code>` or `<b>` or whatever, but rely on the URL detection for link markup.

If you are willing to issue a PR with your simple fix, I think that'd be great.  There's some subtlety to the RE, I suppose, because Perl doesn't like a variable-length `?<!` (lookbehind) assertion.

**Comment by dkolbly** [comment]
Thanks @ilikepi I went ahead and fixed this in #40.  I'm not a perl RE wizard, but just making sure the `src` part was the same length as the `href` part by prefixing a space seemed to work fine.
